www/gamesearchpopup.php

```diff
@@ -204,17 +206,17 @@ function gameSearchPopupDiv()
           <input type=submit name="gameSearchPopupGoBtn"
               id="gameSearchPopupGoBtn" value="Search">
           <script type="text/javascript" nonce="<?php global $nonce; echo $nonce; ?>">
-            gameSearchPopupSearchBox.addEventListener('keypress', function (event) {
-                var result = gameSearchPopupKey(event);
+            gameSearchPopupSearchBox.addEventListener('keypress', async function (event) {
+                var result = await gameSearchPopupKey(event);
                 if (result === false) event.preventDefault();
             });
-            gameSearchPopupSearchBox.addEventListener('keydown', function (event) {
-                var result = gameSearchPopupKey(event);
+            gameSearchPopupSearchBox.addEventListener('keydown', async function (event) {
+                var result = await gameSearchPopupKey(event);
                 if (result === false) event.preventDefault();
             });
-            gameSearchPopupGoBtn.addEventListener('click', function (event) {
+            gameSearchPopupGoBtn.addEventListener('click', async function (event) {
                 event.preventDefault();
-                gameSearchPopupGo();
+                await gameSearchPopupGo();
```

Comment:
This code is _weird_, and I think we should probably clean it up.

In order for `event.preventDefault()` to work, it needs to happen _before_ we `await` a promise. So, the `preventDefault` code here doesn't and can't work.

But, luckily for us, it doesn't _need_ to work! The key handlers are designed to `return false` and thereby `preventDefault` when the user presses the Enter key. (Or the `ch == 10` line feed key?? I don't think any keyboard in modern use has a key that can generate that `charCode`.)

The Enter key would normally submit the form, but on the `editgame` page, we don't want to submit the page's `<form>`.

Luckily, it _doesn't_ submit the form. If you remove the `keypress` and `keydown` listeners entirely, and press Enter, it fires a `click` event on `gameSearchPopupGoBtn` (because it's an `<input type=submit>`), and _that_ prevents default right away.

So, the only purpose of the key handlers is to handle the Escape key, which never even fires `keypress` events, only `keydown` and `keyup` events. Those handlers don't need to be `async` at all. (And they should use `event.code` because `keyCode` is deprecated.)

As for the `click` handler, it doesn't need to be `async` either, because there's no code after `gameSearchPopupGo()`, so there's no point in `await`ing the result.

Reply:
I added the async stuff before I understood that try{} blocks allow use of await.

You're welcome to push more commits to fix these. :)